rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -233,24 +251,52 @@ impl Handler<GetPhyIpFromMeshIp> for TunnelManager {
 }
 
 pub struct GetNeighbors;
-impl Message for GetNeighbors {
-    type Result = Result<Vec<(LocalIdentity, String, IpAddr)>, Error>;
+
+#[derive(Debug)]
+pub struct Neighbor {
+    pub identity: LocalIdentity,
+    pub iface_name: String,
+    pub tunnel_ip: IpAddr,
+}
+
+impl Neighbor {
+    fn new(identity: LocalIdentity, iface_name: String, tunnel_ip: IpAddr) -> Neighbor {
+        Neighbor {
+            identity,
+            iface_name,
+            tunnel_ip,
+        }
+    }
 }
 
+impl Message for GetNeighbors {
+    type Result = Result<Vec<Neighbor>, Error>;
+}
 impl Handler<GetNeighbors> for TunnelManager {
-    type Result = Result<Vec<(LocalIdentity, String, IpAddr)>, Error>;
+    type Result = Result<Vec<Neighbor>, Error>;
 
     fn handle(&mut self, _: GetNeighbors, _: &mut Context<Self>) -> Self::Result {
         let mut res = Vec::new();
-        for obj in self.tunnels.iter() {
-            let tunnel = obj.1;
-            res.push((tunnel.localid.clone(), tunnel.iface_name.clone(), tunnel.ip));
+        for (_, tunnel) in self.tunnels.iter() {
+            res.push(Neighbor::new(
+                tunnel.localid.clone(),
+                tunnel.iface_name.clone(),
+                tunnel.ip,
+            ));
         }
         Ok(res)
     }
 }
 
-pub struct PeersToContact(pub HashMap<IpAddr, Peer>);
+pub struct PeersToContact {
+    pub peers: HashMap<IpAddr, Peer>,
+}
+
+impl PeersToContact {
+    pub fn new(peers: HashMap<IpAddr, Peer>) -> PeersToContact {
```

Comment:
The constructor doesn't do anything

Reply:
Thats whole point of it to add named members just to avoid unnamed members (and accessing them by `.0` `.1` etc.)